workflow/controller/workflowpod.go

```diff
@@ -638,10 +638,10 @@ func (woc *wfOperationCtx) createVolumes(tmpl *wfv1.Template) []apiv1.Volume {
 	}
 	switch woc.getContainerRuntimeExecutor() {
 	case common.ContainerRuntimeExecutorKubelet, common.ContainerRuntimeExecutorK8sAPI, common.ContainerRuntimeExecutorPNS:
-	case common.ContainerRuntimeExecutorEmissary:
-		volumes = append(volumes, volumeVarArgo)
-	default:
+	case common.ContainerRuntimeExecutorDocker:
 		volumes = append(volumes, woc.getVolumeDockerSock(tmpl))
+	default:
+		volumes = append(volumes, volumeVarArgo)
```

Comment:
if I may (been working on #6714 so I saw this too): this changes behaviour because Go's switch-case doesn't fallthrough by default. Before only emissary had /var/run/argo as volume, with this change all except Docker have it as volume. 
This changes anyway in #6714 (all will have that volume + mount) so I'm not sure how important it is to get the correct behaviour here.

Reply:
+1

Reply:
Correct me if I'm wrong. Because go doesn't fall through by default, we have the following and the behaviour is expected

```
pns/k8sapi/kubelet:
   do nothing
docker:
   attach dockerSock
emissary or empty (default):
   attach varArgo
```

Reply:
aah, right. My bad. Sorry about the confusion.